src/util.ts

```diff
@@ -257,3 +290,221 @@ const getCommentLine = (editor: vscode.TextEditor, uuid: string): string => {
       return `// [MermaidChart: ${uuid}]`;
   }
 };
+
+
+/**
+ * Ensures the diagram code has a config block with the given ID.
+ * @param code The original diagram code.
+ * @param diagramId The ID to include in the config block.
+ * @returns The updated diagram code.
+ */
+export function ensureConfigBlock(code: string, diagramId: string): string {
+  const configPattern = /^---\s*config:\s*([\s\S]*?)---/m; // Regex to match the config block
+  const idLine = `    id: ${diagramId}\n`;
+
+  if (configPattern.test(code)) {
+    // If config block exists, update or append the ID
+    const updatedCode = code.replace(configPattern, (match, configContent) => {
+      if (configContent.includes("id:")) {
+        // Update the existing ID
+        return match.replace(/id:\s*.+/m, idLine.trim());
+      } else {
+        // Append the ID
+        return match.replace("config:", `config:\n${idLine}`);
+      }
+    });
+
+    // Check for and remove empty lines below the id field
+    return updatedCode.replace(/(id:\s*.+)\n\s*\n/m, "$1\n");
+  } else {
+    // If config block doesn't exist, add the entire block
+    const configBlock = `---\nconfig:\n${idLine}---\n\n`;
```

Comment:
This logic would not work, there could be diagrams with frontmatter, without the config key, and just the title or some other key. Then this logic would add two frontmatters to the code, breaking it.